unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarsLODController.cs

```diff
@@ -0,0 +1,147 @@
+                if (!lodGO.activeSelf)
+                    continue;
+
+                if (applyVerticalMovement)
+                    lodGO.transform.localPosition = new Vector3(lodGO.transform.localPosition.x, (lodGO.transform.localPosition.y > LODS_LOCAL_Y_POS ? LODS_LOCAL_Y_POS : LODS_LOCAL_Y_POS + LODS_VERTICAL_MOVEMENT), lodGO.transform.localPosition.z);
+
+                Vector3 previousForward = lodGO.transform.forward;
+                Vector3 lookAtDir = (lodGO.transform.position - CommonScriptableObjects.cameraPosition).normalized;
+
+                lookAtDir.y = previousForward.y;
+
+                lodGO.transform.forward = lookAtDir;
+            }
+
+            if (applyVerticalMovement)
+                lastLODsVerticalMovementTime = Time.timeSinceLevelLoad;
+        }
+
+        void OnMainPlayerReposition(Vector3 newPos, Vector3 previousPos) { UpdateAllLODs(); }
+
+        public void UpdateAllLODs()
+        {
+            if (!enabled)
+                return;
+
+            SortedList<float, AvatarShape> closeDistanceAvatars = new SortedList<float, AvatarShape>();
+            foreach (AvatarShape avatar in avatarsList)
+            {
+                float distanceToPlayer = Vector3.Distance(CommonScriptableObjects.playerUnityPosition.Get(), avatar.transform.position);
+                bool isInLODDistance = distanceToPlayer >= lodDistance;
+
+                if (isInLODDistance)
+                    ToggleLOD(avatar.avatarRenderer, true);
+                else
+                    closeDistanceAvatars.Add(distanceToPlayer, avatar);
+            }
+
+            int closeDistanceAvatarsCount = closeDistanceAvatars.Count;
+            AvatarShape currentAvatar;
+            for (var i = 0; i < closeDistanceAvatarsCount; i++)
+            {
+                currentAvatar = closeDistanceAvatars.Values[i];
+                bool isLOD = i >= maxNonLODAvatars;
+
+                ToggleLOD(currentAvatar.avatarRenderer, isLOD);
+            }
+        }
+
+        private void ToggleLOD(AvatarRenderer avatarRenderer, bool enabled)
+        {
+            avatarRenderer.lodRenderer.gameObject.SetActive(enabled);
+            avatarRenderer.SetVisibility(!enabled); // TODO: Resolve coping with AvatarModifierArea regarding this toggling
+        }
+
+        public void RegisterAvatar(AvatarShape newAvatar)
+        {
+            if (!enabled || avatarsList.Contains(newAvatar))
+                return;
+
+            avatarsList.Add(newAvatar);
+
+            // TODO: Find a way to get this behaviour but unsubscribing on RemoveAvatar()
+            // DecentralandEntity already nullifies this event on cleanup, maybe that's good enough?
+            // Worst case scenario we'll have to refactor the DCLEntity.OnTransformChange event to report which entity triggered the event
+            newAvatar.entity.OnTransformChange += (object newTransformModel) =>
+            {
+                UpdateAllLODs();
```

Comment:
This shouldn't be very unperformant? If this method iterated over all avatars, this operation approaches O(n*n). It isn't better to just mark dirty and have all the lods updated a single time per frame?

Reply:
True that, better to update the LODs on every frame, even if it triggers unneeded updates, instead of on every character+avatars movement. Done.